lua/tests/helpers.lua

```diff
@@ -116,11 +122,130 @@ local function has_all(_, arguments)
   return true
 end
 
+local prev_buf_max = -1
+local prev_win_max = -1
+local prev_buf
+local prev_win
+
+local function change_infoview(state, _)
+  local this_infoview = infoview.get_current_infoview()
+  local buf = this_infoview.bufnr
+  local win = this_infoview.window
+  local result =
+    ((not buf and not this_infoview.is_open) or (buf and prev_buf ~= buf and this_infoview.is_open
+      and vim.api.nvim_buf_is_valid(buf))) and
+    ((not win and not this_infoview.is_open) or (win and prev_win ~= win and this_infoview.is_open
+      and vim.api.nvim_win_is_valid(win))) and
+    helpers.get_num_wins() == last_num_wins
+  prev_buf = buf
+  prev_win = win
+  state.failure_message = table.concat({
+    "Failed to change: ",
+    ("prev_buf: %s, buf: %s, buf valid: %s"):format(vim.inspect(prev_buf), vim.inspect(buf),
+      vim.inspect(buf and vim.api.nvim_buf_is_valid(buf))),
+    ("prev_win: %s, win: %s, win valid: %s"):format(vim.inspect(prev_win), vim.inspect(win),
+      vim.inspect(win and vim.api.nvim_win_is_valid(win))),
+    "is_open: " .. vim.inspect(this_infoview.is_open),
+    ("num_wins: %d, last_num_wins: %d"):format(helpers.get_num_wins(), last_num_wins)
+  }, "\n")
+
+  return result
+end
+
+local function open_infoview(state, arguments)
```

Comment:
Can we try to avoid passing boolean flags to functions? The tests below are quite hard to read because it's hard to know what `assert.open_infoview(false, true)` even means.